os/tools/mkbinheader.py

```diff
@@ -19,6 +19,69 @@
 import os
 import sys
 import struct
+
+# Temporary file to estimate the static RAM size.
+STATIC_RAM_ESTIMATION = 'temp_static_ram_estimation_file'
+
+def roundup_power_two(size):
+    size = size - 1
+    size |= size >> 1
+    size |= size >> 2
+    size |= size >> 4
+    size |= size >> 8
+    size |= size >> 16
+    size = size + 1
+
+    return size
+
+
+def get_static_ram_size(bin_type):
+    textsize = 0
+    rosize = 0
+    datasize = 0
+    bsssize = 0
+    ram_fp = open(STATIC_RAM_ESTIMATION, 'rb')
+    if bin_type == BIN :
+        # First line is not used for calculating RAM size. So throw away it.
+        ram_fp.readline()
+        ram_array = ram_fp.readline()
+        size_array = ram_array.split('\t')
+        static_ram_size = size_array[SIZE_CMD_SUMMATION_INDEX]
+    elif bin_type == ELF :
+        line = ram_fp.readline()
+        while line:
+            words = line.split('.')
+            if len(words) > 1:
+                words = words[1].split()
+                if len(words) > 1:
+                    section = words[0]
+                    size = int(words[4], 16)
+                    if section == 'text':
+                        textsize = size
+                    elif section == 'rodata':
+                        rosize = size
+                    elif section == 'data':
+                        datasize = size
+                    elif section == 'bss':
+                        bsssize = size
+                        break
+            line = ram_fp.readline()
+        textsize = roundup_power_two(textsize)
+        # If CONFIG_OPTIMIZE_APP_RELOAD is enabled, then we will make a copy
+        # of the data section inside the ro section and it will be used in
+        # reload time. So, we add datasize to rosize to make place for data section.
+        rosize = rosize + datasize;
```

Comment:
I made a python code for checking the config.
```
import string

def check_optimize_config(file_name):
	with open(file_name, 'r+') as f:
		lines = f.readlines()
        
	return any([True if 'CONFIG_OPTIMIZE_APP_RELOAD=y' in line and not line.startswith('#') else False for line in lines ])

if check_optimize_config('../.config') == True:
	print("CONFIG_OPTIMIZE_APP_RELOAD is enabled.")
else:
	print("CONFIG_OPTIMIZE_APP_RELOAD is disabled.")
```
You can use this with replacing print to your functionality.

Reply:
Thank you @jeongchanKim 
I have used your code. I had to make a small change for obtaining the .config path. 
It is working well now.